pages/api/auth/signin.js

```diff
@@ -0,0 +1,51 @@
+require("dotenv").config();
+import bcrypt from "bcrypt";
+import sql from "~/lib/postgres";
+import * as yup from "yup";
+import jwt from "jsonwebtoken";
+
+const schema = yup.object().shape({
+  email: yup.string().email().required(),
+  hash_password: yup.string().required(),
+});
+
+const handler = async ({ method, body: { email, password } }, res) => {
+  if (method === "POST") {
+    const user = {
+      email: email.toLowerCase(),
+      hash_password: password,
+    };
+    const schemaValid = await schema.isValid(user);
+    if (schemaValid) {
+      const userExists = await sql`SELECT * FROM users WHERE email=${email}`;
+      if (userExists.count > 0) {
+        try {
+          const userValid = await bcrypt.compare(
+            password,
+            userExists[0].hash_password
+          );
+          if (userValid) {
+            const jsonToken = await jwt.sign(
+              user,
+              process.env.ACCESS_TOKEN_SECRET
+            );
+            return res.send(jsonToken);
+            return res.status(200).json({ message: "Success" });
+          } else {
+            return res
+              .status(400)
+              .json({ message: "Email or password incorrect" });
+          }
+        } catch (error) {
+          res.status(500).json({ errorMessage: error });
+        }
+      } else {
+        return res.status(400).send({ message: "Email or password incorrect" });
```

Comment:
A little hard to be sure I'm talking about the correct else here, but I think this goes with the `if(userExists.count...)` block. If we hit this else, isn't it the case that the user isn't found? It might make sense to have this message show a message something more along those lines. To me it seems that this block is hit if you don't find the email